src/commissaire_http/bus/__init__.py

```diff
@@ -0,0 +1,196 @@
+        self.connection = None
+        self._channel = None
+        self._exchange = None
+        self.exchange_name = exchange_name
+        self.connection_url = connection_url
+        self.qkwargs = qkwargs
+
+    @property
+    def init_kwargs(self):
+        """
+        Returns the initializing kwargs for this instance.
+
+        :returns: the initializing kwargs for this instance.
+        :rtype: dict
+        """
+        return {
+            'exchange_name': self.exchange_name,
+            'connection_url': self.connection_url,
+            'qkwargs': self.qkwargs,
+        }
+
+    def connect(self):
+        """
+        'Connects' to the bus.
+
+        :returns: The same instance.
+        :rtype: commissaire_http.bus.Bus
+        """
+        self.connection = Connection(self.connection_url)
+        self._channel = self.connection.channel()
+        self._exchange = Exchange(
+            self.exchange_name, type='topic').bind(self._channel)
+        self._exchange.declare()
+
+        # Create queues
+        self._queues = []
+        for kwargs in self.qkwargs:
+            queue = Queue(**kwargs)
+            queue.exchange = self._exchange
+            queue = queue.bind(self._channel)
+            self._queues.append(queue)
+            self.logger.debug('Created queue {}'.format(queue.as_dict()))
+
+        # Create producer for publishing on topics
+        self.producer = Producer(self._channel, self._exchange)
+        self.logger.debug('Bus connection finished')
+        return self
+
+    # TODO: Everything under this line needs to move to a common module
+    #       shared between service and http
+
+    @classmethod
+    def create_id(cls):  # pragma: no cover
+        """
+        Creates a new unique identifier.
+
+        :returns: A unique identification string.
+        :rtype: str
+        """
+        return str(uuid.uuid4())
+
+    def respond(self, queue_name, id, payload, **kwargs):  # pragma: no cover
+        """
+        Sends a response to a simple queue. Responses are sent back to a
+        request and never should be the owner of the queue.
+
+        :param queue_name: The name of the queue to use.
+        :type queue_name: str
+        :param id: The unique request id
+        :type id: str
+        :param payload: The content of the message.
+        :type payload: dict
+        :param kwargs: Keyword arguments to pass to SimpleQueue
+        :type kwargs: dict
+        """
+        self.logger.debug('Sending response for message id "{}"'.format(id))
+        send_queue = self.connection.SimpleQueue(queue_name, **kwargs)
+        jsonrpc_msg = {
+            'jsonrpc': "2.0",
+            'id': id,
+            'result': payload,
+        }
+        self.logger.debug('jsonrpc msg: {}'.format(jsonrpc_msg))
+        send_queue.put(jsonrpc_msg)
+        self.logger.debug('Sent response for message id "{}"'.format(id))
+        send_queue.close()
```

Comment:
That's the plan! I want to move some of the items used in both places in to the ``comissaire`` package.